src/ray/util/compat.cc

```diff
@@ -0,0 +1,102 @@
+// Copyright 2025 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/util/compat.h"
+
+#include <cstring>
+
+#include "ray/util/logging.h"
+
+#if defined(__APPLE__) || defined(__linux__)
+#include <unistd.h>
+#if defined(HAVE_FDATASYNC) && !HAVE_DECL_FDATASYNC
+extern int  fdatasync(int fildes);
+#endif
```

Comment:
is this needed?

Reply:
It comes from @aslonnie 's link: https://postgrespro.com/list/thread-id/2525589